src/modules/commander/failsafe/failsafe.cpp

```diff
@@ -472,6 +472,14 @@ void Failsafe::checkStateAndMode(const hrt_abstime &time_us, const State &state,
 		CHECK_FAILSAFE(status_flags, local_position_accuracy_low, ActionOptions(Action::RTL));
 	}
 
+	if (state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_TAKEOFF ||
+	    state.user_intended_mode == vehicle_status_s::NAVIGATION_STATE_AUTO_RTL) {
+		CHECK_FAILSAFE(status_flags, navigator_failure, ActionOptions(Action::Land).clearOn(ClearCondition::OnModeChangeOrDisarm));
+
+	} else {
+		CHECK_FAILSAFE(status_flags, navigator_failure, ActionOptions(Action::Hold).clearOn(ClearCondition::OnModeChangeOrDisarm));
```

Comment:
ideally, the navigator can propose the action instead of them being hardcoded here. Because depending on the issue, you might want to RTL instead of hold. Also you do not have to make here the distinction between the modes anymore. Would propaply need to set the desired mode in the navigator_failure instead of a boolean. But depending on the amount of work the changes would need, i am also fine with this for now.

Reply:
I thought that the failsafe action should always be chosen by Commander. Navigator should just be the "follower" and reports back to Commander, no?

Reply:
Yeah, you are right, i thought that a nexternal mode can suggest an action, but it looks like not, sorry. Seems you can only defer the failsafe as external mode.